mmv1/templates/tgc_next/cai2hcl/resource_converter.go.tmpl

```diff
@@ -109,11 +109,9 @@ func (c *{{ $.ResourceName -}}Cai2hclConverter) convertResourceData(asset caiass
 	}
 {{ end}}
 
-{{ range $param := $.UserParameters }}
-{{/* Attempt to parse all parameters from asset name. */}}
-	hclData["{{ underscore $param.Name }}"] = utils.ParseFieldValue(asset.Name, "{{ underscore (plural $param.Name) }}")
-{{ end }}
-{{- if $.HasProject -}}
+{{/* Attempt to parse all self-link parameters from asset name. */}}
+	utils.ParseUrlParamValuesFromAssetName(asset.Name, "{{ $.SelfLinkUri }}", hclData)
```

Comment:
When generating the asset name in the tfplan2cai converter [template](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/templates/tgc_next/tfplan2cai/resource_converter.go.tmpl#L56), we call the function `CaiAssetNameTemplate`. We can reuse the logic in that [function](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/api/resource.go#L1872-L1881) instead of `SelfLinkUri`. `SelfLinkUri` may not cover all of the cases.

Reply:
Done.